File: Src/Engine/Editor/SceneGraph/RootGraphNode.cpp

#include "RootGraphNode.h"

#include "Editor/GUI/Tree/ActorTreeNode.h"

namespace SE::Editor
{
	::SE::Transform RootGraphNode::__GetTransform()
	{
		return Transform::Identity;
	}

	void RootGraphNode::__SetTransform(::SE::Transform value)
	{

	}

	RootGraphNode::RootGraphNode() : ActorGraphNode(nullptr, UID::New())
	{
		_treeNode->AutoFocus = false;
	}
	RootGraphNode::RootGraphNode(UID id) : ActorGraphNode(nullptr, id)
	{
		_treeNode->AutoFocus = false;
	}
} // SE

File: Src/Engine/Editor/SceneGraph/RootGraphNode.cs
using System;
using System.Collections.Generic;

namespace SE.Editor.SceneGraph
{
    /// <summary>
    /// Root of the editor scene graph.
    /// </summary>
    public abstract class RootGraphNode : ActorGraphNode
    {
        protected RootGraphNode(SceneGraphFactory factory)
            : this(factory, Guid.NewGuid())
        {
        }

        protected RootGraphNode(SceneGraphFactory factory, Guid id)
            : base(factory, null, id)
        {
            TreeNode.AutoFocus = false;
        }

        public override string Name => "Root";
        public override RootGraphNode Root => this;
        public override SceneGraphNode? ParentScene => null;

        public override Transform Transform
        {
            get => SE.Transform.Identity;
            set { }
        }

        public abstract IReadOnlyList<ScenesGraphNode> Selection { get; }
        public abstract void Spawn(SE.Actor actor, SE.Actor? parent);

        public virtual void OnActorChildNodesDispose(ActorGraphNode node)
        {
        }
    }
}
